services/ux-backend/handlers/removecephclient/handler.go

```diff
@@ -0,0 +1,128 @@
+package removecephclient
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	ocsv1alpha1 "github.com/red-hat-storage/ocs-operator/api/v4/v1alpha1"
+	controllers "github.com/red-hat-storage/ocs-operator/v4/controllers/storageconsumer"
+	"github.com/red-hat-storage/ocs-operator/v4/services/ux-backend/handlers"
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func HandleMessage(w http.ResponseWriter, r *http.Request, cl client.Client, consumerName string) {
+	switch r.Method {
+	case "POST":
+		handlePost(r.Context(), w, cl, consumerName)
+	default:
+		handleUnsupportedMethod(w, r)
+	}
+}
+
+func handlePost(ctx context.Context, w http.ResponseWriter, cl client.Client, consumerName string) {
+	klog.Info("POST method on /remove-ceph-client endpoint is invoked")
+	w.Header().Set("Content-Type", handlers.ContentTypeTextPlain)
+
+	if message, err := removeCephZombieResources(ctx, cl, consumerName); err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		w.Header().Set("Content-Type", handlers.ContentTypeTextPlain)
+
+		if _, err := w.Write([]byte("Failed to delete resources")); err != nil {
+			klog.Errorf("failed write data to response writer, %v", err)
+		}
+	} else {
+		klog.Info(message)
+		w.WriteHeader(http.StatusOK)
+		w.Header().Set("Content-Type", handlers.ContentTypeTextPlain)
+
+		if _, err = w.Write([]byte(message)); err != nil {
+			klog.Errorf("failed write data to response writer: %v", err)
+		}
+	}
+
+}
+
+func handleUnsupportedMethod(w http.ResponseWriter, r *http.Request) {
+	klog.Info("Only POST method should be used to send data to this endpoint /remove-ceph-client")
+	w.WriteHeader(http.StatusMethodNotAllowed)
+	w.Header().Set("Content-Type", handlers.ContentTypeTextPlain)
+	w.Header().Set("Allow", "POST")
+
+	if _, err := w.Write([]byte(fmt.Sprintf("Unsupported method : %s", r.Method))); err != nil {
+		klog.Errorf("failed to write data to response writer: %v", err)
+	}
+}
+
+func removeCephZombieResources(ctx context.Context, cl client.Client, consumerName string) (string, error) {
```

Comment:
should a http request perform all these operations or let the storageconsumer know it's time for force removal? My reasoning is:

if everything is successful the entry will be removed from UI, however this can't be guaranteed just by placing annotations on ceph resources, the backend controllers should work on cleanup those resources.

suppose if we didn't annotate svg but did for rns still we failed the cleanup but did partially, so why not reduce the work for the backend by just placing an annotation on storageconsumer and modify storageconsumer controller to perform the cleanup?

moreover the client here isn't backed by a cache and every http call will fetch these resources from api server. wdyt?

Reply:
Yes idea looks better than the one I implemented, Do you think we should add the same annotation that we are adding on SVG/RNS or should we introduce a new one?